crates/sui-core/src/unit_tests/pay_sui_tests.rs

```diff
@@ -335,11 +324,8 @@ async fn test_pay_all_sui_failure_insufficient_gas_one_input_coin() {
     let res = execute_pay_all_sui(vec![&coin1], recipient, sender, sender_key, 2000).await;
 
     assert_eq!(
-        res.txn_result
-            .unwrap_err()
-            .collapse_if_single_transaction_input_error()
-            .unwrap(),
-        &SuiError::GasBalanceTooLowToCoverGasBudget {
+        UserInputError::try_from(res.txn_result.unwrap_err()).unwrap(),
```

Comment:
you could probably do some work with `match!` to avoid the unwraps and conversions, but this is mostly fine as its just in test code.